include/wpe/gamepad.h

```diff
@@ -0,0 +1,285 @@
+ * @provider: opaque libwpe's representation of gamepad provider.
+ *
+ * Called by WPEWebkit to stop @provider monitoring for gamepad
+ * devices.
+ */
+WPE_EXPORT
+void
+wpe_gamepad_provider_stop(struct wpe_gamepad_provider*);
+
+/**
+ * wpe_gampepad_provider_get_view_backend_for_gamepad:
+ * @provider: opaque libwpe's representation of gamepad provider.
+ * @gamepad: opaque libwep's representation of gampead
+ *
+ * Ask @provider for the view to where @gamepad is attached.
+ */
+WPE_EXPORT
+struct wpe_view_backend *
+wpe_gamepad_provider_get_view_backend_for_gamepad(struct wpe_gamepad_provider *, struct wpe_gamepad *);
+
+/**
+ * wpe_gamepad_provider_dispatch_gamepad_connected:
+ * @provider: opaque libwpe's representation of gamepad provider.
+ * @gamepad_id: application's gamepad device identifier.
+ *
+ * Method called by application (gamepad implementator) to inform
+ * WPEWebKit that a new gamepad device is plugged.
+ *
+ * Since: 1.12
+ */
+WPE_EXPORT
+void
+wpe_gamepad_provider_dispatch_gamepad_connected(struct wpe_gamepad_provider*, unsigned);
+
+/**
+ * wpe_gamepad_provider_dispatch_gamepad_disconnected:
+ * @provider: opaque libwpe's representation of gamepad provider.
+ * @gamepad_id: application's gamepad device identifier.
+ *
+ * Method called by application (gamepad implementator) to inform
+ * WPEWebKit that a plugged gamepad device has been unplugged.
+ *
+ * Since: 1.12
+ */
+WPE_EXPORT
+void
+wpe_gamepad_provider_dispatch_gamepad_disconnected(struct wpe_gamepad_provider*, unsigned);
+
+/**
+ * wpe_gamepad_create:
+ * @gamepad_id: opaque application's representation of gamepad provider.
+ *
+ * Method called by WPEWebKit to create an internal representation of a gamepade device.
+ *
+ * Returns: opaque libwpe's representation of a gamepad device.
+ */
+WPE_EXPORT
+struct wpe_gamepad*
+wpe_gamepad_create(unsigned);
+
+/**
+ * wpe_gamepad_destroy:
+ * @gamepad: opaque libwpe's representation of gamepad.
+ *
+ * Called by WPEWebkit to free the internal resources used by @gamepad
+ * object.
+ */
+WPE_EXPORT
+void
+wpe_gamepad_destroy(struct wpe_gamepad*);
+
+/**
+ * wpe_gamepad_provider_set_client:
+ * @gamepad: opaque libwpe's representation of gamepad.
+ * @client_interface: WPEWebKit callbacks for gamepad devices events.
+ * @client_data: WPEWebkit closure data.
+ *
+ * Sets WPEWebKit's callbacks for events in @gamepad.
+ */
+WPE_EXPORT
+void
+wpe_gamepad_set_client(struct wpe_gamepad*, struct wpe_gamepad_client_interface*, void*);
+
+/**
+ * wpe_gamepad_get_id:
+ * @gamepad: opaque libwpe's representation of gamepad.
+ *
+ * Gets the identification string for the @gamepad.
+ */
+WPE_EXPORT
+const char*
+wpe_gamepad_get_id(struct wpe_gamepad*);
+
+/**
+ * wpe_gamepad_dispatch_button_changed:
+ * @gamepad: opaque gamepad object.
+ * @button: the standard button that changed its state.
+ * @pressed: the new state: %TRUE pressed, otherwise released.
+ *
+ * Callbacks WPEWebkit to update the status of @button.
```

Comment:
My brain is having trouble parsing this. Does it mean “Report to WebKit a change in the status of *button*”?